module/VuFind/src/VuFind/Config/Handler/Ini.php

```diff
@@ -130,4 +140,157 @@ protected function getMergeCallback(array $overrideFullSections, bool $mergeArra
             return $parentConfig;
         };
     }
+
+    /**
+     * Write configuration to a specific location.
+     *
+     * @param ConfigLocationInterface  $destinationLocation Destination location for the config
+     * @param array|string             $config              Config to write
+     * @param ?ConfigLocationInterface $baseLocation        Location of a base configuration that can provide additional
+     * structure (e.g. comments)
+     *
+     * @return void
+     */
+    public function writeConfig(
+        ConfigLocationInterface $destinationLocation,
+        array|string $config,
+        ?ConfigLocationInterface $baseLocation
+    ): void {
+        if (!is_array($config)) {
+            throw new ConfigException('Ini handler can only write array config.');
+        }
+
+        // If target file already exists, back it up:
+        $outfile = $destinationLocation->getPath();
+        $this->backupFile($outfile);
+
+        $comments = [];
+        if ($baseLocation !== null) {
+            $comments = $this->extractComments($baseLocation->getPath());
+        }
+        $writer = $this->getConfigWriter($outfile, $comments, $comments);
```

Comment:
I think a bug was introduced here:

```suggestion
        $writer = $this->getConfigWriter($outfile, $config, $comments);
```